dbt/adapters/dremio/api/parameters.py

```diff
@@ -47,21 +47,25 @@ def build(cls, credentials: DremioCredentials):
             and credentials.cloud_project_id is not None
         ):
             return CloudParametersBuilder(
-                cls._build_dremio_authentication(credentials=credentials),
-                credentials.cloud_host,
-                credentials.cloud_project_id,
+                authentication=cls._build_dremio_authentication(
+                    credentials=credentials
+                ),
+                cloud_host=credentials.cloud_host,
+                cloud_project_id=credentials.cloud_project_id,
             )
-        elif (
+        if (
```

Comment:
elif actually results in faster performance because the interpreter doesn't have to reevaluate statements, that's not too significant in this case, but I think it's better code keeping it as elif.

Reply:
Since the first if contains a return statement, the interpreter will never reach your elif line if the constraint is satisfied. It's redundant to use elif. The use of else or elif becomes redundant becomes redundant in this case.

Reply:
Yes but it's also a readability aspect - using if/elif/else vs just if explicitly shows that each block is mutually exclusive. 

https://stackoverflow.com/questions/22782785/when-to-use-if-vs-elif-in-python

Reply:
I can agree with that, but I changed it based on a suggestion from pylint:

no-else-return (R1705):
Unnecessary "%s" after "return", %s Used in order to highlight an unnecessary block of code following an if containing a return statement. As such, it will warn when it encounters an else following a chain of ifs, all of them containing a return statement.

Reply:
That warning seems to not be liked by most people:
https://github.com/SoCo/SoCo/issues/500
https://stackoverflow.com/questions/55561042/what-is-the-real-benefit-of-obeying-r1705-in-pylint-is-the-code-really-safer

This is definitely a style question though and we should discuss as a team.